frontend/src/pages/CachePage/CachePage.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getCache } from "../../api/geocaches";
+import { Geocache } from "../../model/Geocache";
+import NavBar from "../../components/NavBar";
+import "./CachePage.scss";
+
+function CachePage() {
+    let { cacheId } = useParams();
+    const [cache, setCache] = useState<Geocache>({
+        referenceCode: "",
+        name: "",
+        placedDate: "",
+        publishedDate: "",
+        type: "",
+        size: "",
+        postedCoordinates: {
+            latitude: 0,
+            longitude: 0
+        },
+        lastVisitedDate: "",
+        isPremiumOnly: false,
+        shortDescription: "",
+        longDescription: "",
+        hints: ""
+    });
+
+    if (cacheId?.at(0) == ":") { cacheId = cacheId.slice(1); }
+    useEffect(() => {
+        getCache(cacheId).then(cacheResult => {
+            setCache(cacheResult);
+        });
+    });
```

Comment:
This hook actually runs infinitely, if you don't give it an [array of dependencies](https://devtrium.com/posts/dependency-arrays). You can see if you open chrome's networking tab, that this hook sends requests infinitely to the backend. Also it should be checked that there is a valid id before doing the api call. These issues can be fixed by writing this useEffect like this: 

    useEffect(() => {
        if(cacheId !== undefined){
            getCache(cacheId).then(cacheResult => {
                setCache(cacheResult);
            });
        }
    }, [cacheId]);

Reply:
Ok